src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -1277,64 +1277,72 @@ private static unsafe int ApplyMask(Span<byte> toMask, int mask, int maskIndex)
         {
             Debug.Assert(maskIndex < sizeof(int));
 
-            int maskShift = maskIndex * 8;
-            int shiftedMask = (int)(((uint)mask >> maskShift) | ((uint)mask << (32 - maskShift)));
-
-            // Try to use SIMD.  We can if the number of bytes we're trying to mask is at least as much
-            // as the width of a vector and if the width is an even multiple of the mask.
-            if (Vector.IsHardwareAccelerated &&
-                Vector<byte>.Count % sizeof(int) == 0 &&
-                toMask.Length >= Vector<byte>.Count)
-            {
-                Vector<byte> maskVector = Vector.AsVectorByte(new Vector<int>(shiftedMask));
-                Span<Vector<byte>> toMaskVector = MemoryMarshal.Cast<byte, Vector<byte>>(toMask);
-                for (int i = 0; i < toMaskVector.Length; i++)
+            fixed (byte* toMaskBeg = toMask)
+            {
+                byte* toMaskPtr = toMaskBeg;
+                byte* toMaskEnd = toMaskBeg + toMask.Length;
+                byte* maskPtr = (byte*)&mask;
+
+                if (toMaskEnd - toMaskPtr < sizeof(int))
                 {
-                    toMaskVector[i] ^= maskVector;
+                    goto small;
                 }
 
-                // Fall through to processing any remaining bytes that were less than a vector width.
-                toMask = toMask.Slice(Vector<byte>.Count * toMaskVector.Length);
-            }
+                // align our pointer to sizeof(int)
 
-            // If there are any bytes remaining (either we couldn't use vectors, or the count wasn't
-            // an even multiple of the vector width), process them without vectors.
-            int count = toMask.Length;
-            if (count > 0)
-            {
-                fixed (byte* toMaskPtr = &MemoryMarshal.GetReference(toMask))
+                while ((ulong)toMaskPtr % sizeof(int) != 0)
                 {
-                    byte* p = toMaskPtr;
+                    *toMaskPtr++ ^= maskPtr[maskIndex];
+                    maskIndex = (maskIndex + 1) & 3;
+                }
```

Comment:
I think it'd be good to add an assert here:
```C#
Debug.Assert(toMaskPtr < toMaskEnd);
```
in case something ever got messed up with the earlier `if` check that's guaranteeing this.  When it comes to unsafe code and pointer manipulation, we try to be extra careful, especially in networking.